src/login.c

```diff
@@ -725,7 +725,7 @@ int main (int argc, char **argv)
 			          sizeof (loginprompt),
 			          _("%s login: "), hostn);
 		} else {
-			strncpy (loginprompt, _("login: "),
+			strlcpy (loginprompt, _("login: "),
 			         sizeof (loginprompt));
```

Comment:
Actually, this one is funny.  The GCC warning is nonsense, since the only correct use of `strncpy(3)` (the only one in which `strlcpy(3)` isn't the right call) requires that the third argument is exactly the size of the first argument (so it's the destination size).  In the updated manual page for strncpy(3), this is described.  `strncpy(3)` is rightfully used when used copying a string into a fixed-width buffer, for example, a structure that is later sent through the network.

However, in this case we're not doing that, and we should really call `strlcpy(3)`.  The original call was very bogus, since it didn't terminate the string.

Good catch!

I think I didn't catch this one in my previous patches because I was grepping for strncpy() followed by setting `'\0'`.